lib/npm-installation-manager.ts

```diff
@@ -42,19 +42,22 @@ export class NpmInstallationManager implements INpmInstallationManager {
 		return path.join(this.getCacheRootPath(), packageName, version, "package");
 	}
 
-	public addToCache(packageName: string, version: string): IFuture<void> {
+	public addToCache(packageName: string, version: string): IFuture<any> {
 		return (() => {
 			let cachedPackagePath = this.getCachedPackagePath(packageName, version);
+			let cachedPackageData: any;
 			if(!this.$fs.exists(cachedPackagePath).wait() || !this.$fs.exists(path.join(cachedPackagePath, "framework")).wait()) {
-				this.addToCacheCore(packageName, version).wait();
+				cachedPackageData = this.addToCacheCore(packageName, version).wait();
 			}
-
-			if(!this.isShasumOfPackageCorrect(packageName, version).wait()) {
+			let realVersion = (cachedPackageData && cachedPackageData.version) || version;
```

Comment:
You can preferably add a comment here explaining why the `cachedPackageData`'s version prevails over `version` if both are present